actionview/lib/action_view/routing_url_for.rb

```diff
@@ -147,5 +135,9 @@ def optimize_routes_generation? #:nodoc:
       def _generate_paths_by_default
         true
       end
+
+      def get_only_path(host)
+        host.nil? ? _generate_paths_by_default : false
```

Comment:
@rafaelfranca @kaspth Great. 
I will change method name and will be checking for return value.